### PR TITLE
[wasm] Throw an exception in some blocking operations in non-threaded mode.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by vargaz** [comment]
@monojenkins build failed

File: mono/metadata/threads.c [hunk 1]
```diff
@@ -2314,6 +2314,22 @@ ves_icall_System_Threading_WaitHandle_Wait_internal (gpointer *handles, gint32 n
 
 	MonoW32HandleWaitRet ret;
 
+#ifdef DISABLE_THREADS
+	if (numhandles == 1 && timeout == MONO_INFINITE_WAIT) {
```
Comment (BrzVlad) [inline mono/metadata/threads.c]: the `numhandles == 1` seems that is not needed

Comment (vargaz) [reply mono/metadata/threads.c]: It fails some tests without it. So this is just a hack to detect some deadlocking apps, it doesn't have to catch all cases.

Comment (BrzVlad) [reply mono/metadata/threads.c]: I guess the loop computation of signaled below is overkill then

**Comment by vargaz** [comment]
@monojenkins build failed

File: mono/metadata/threads.c (hunk 1 above)
Comment (vargaz) [reply mono/metadata/threads.c]: Already signalled/owned events can be waited for since the wait will end immediately.

**Pull request merged**